File: applicatie/src/main/java/com/vdzon/administratie/rest/administratie/AdministratieResource.java
package com.vdzon.administratie.rest.administratie;

import com.vdzon.administratie.util.JsonUtil;
import spark.Spark;

import javax.inject.Inject;

public class AdministratieResource {

    @Inject
    public AdministratieResource(AdministratieService administratieService) {
        Spark.post("/rest/administratie", administratieService::putAdministratie, JsonUtil.json());
    }
}

File: applicatie/src/main/java/com/vdzon/administratie/rest/administratie/AdministratieResource.kt
package com.vdzon.administratie.rest.administratie

import com.vdzon.administratie.util.JsonUtil
import spark.Route
import spark.Spark

import javax.inject.Inject

class AdministratieResource
@Inject
constructor(administratieService: AdministratieService) {
    init {
        Spark.post("/rest/administratie", Route { req, res -> administratieService.putAdministratie(req, res) }, JsonUtil.json())
    }
}
